NuRadioReco/modules/LOFAR/stationPulseFinder.py

```diff
@@ -299,11 +299,21 @@ def run(self, event, detector):
             station.set_parameter(stationParameters.cr_dominant_polarisation,
                                   detector.get_antenna_orientation(station_id, ant_same_orientation[dominant_pol][0]))
 
-            # Check if the station has a strong enough signal
-            signal_window = [int(pulse_window_start), int(pulse_window_end)]
-            noise_window = [0, int(pulse_window_start - self.__noise_away_from_pulse)]
-
-            self._check_station_triggered(station, position_array, ant_same_orientation, signal_window, noise_window)
+            # # Check if the station has a strong enough signal
+            # signal_window = [int(pulse_window_start), int(pulse_window_end)]
+            # noise_window = [0, int(pulse_window_start - self.__noise_away_from_pulse)]
+
+            # there was a bug with the windows for the SNR calculation, where strong signals had low SNR and vice versa. 
+            # Since the pulse should be somewhere around the middle of the trace, we use a fixed noise window (without the tapered edges of the trace). 
+            # For the signal window, the whole trace is used since we need the maximum. 
+            # Narrowing this down may increase performance and may be worth looking into in the future.
+            self._check_station_triggered(
+                station, 
+                position_array, 
+                ant_same_orientation, 
+                signal_window=[0, -1], 
```

Comment:
Why can't we keep the window calculation as before? I thought we only needed to change the noise windows to not include the tapered edges?

Reply:
Apparently not because then strong signals would still have high SNRs. But I found that strong signals had low SNRs, so there must be some issue with the peak amplitude. I suspect this comes from the signal window.

Reply:
I'm sorry, but it is not clear to me whether you tried to keep the signal window as before, and only change the noise window?